opsmngr/backup_administrator.go

```diff
@@ -0,0 +1,874 @@
+// Copyright 2019 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	backupAdministratorBlockstoreBasePath                    = "admin/backup/snapshot/mongoConfigs"
+	backupAdministratorFileSystemStoreConfigurationsBasePath = "admin/backup/snapshot/fileSystemConfigs"
+	backupAdministratorS3BlockstoreBasePath                  = "admin/backup/snapshot/s3Configs"
+	backupAdministratorOplogBasePath                         = "admin/backup/oplog/mongoConfigs"
+	backupAdministratorSyncBasePath                          = "admin/backup/sync/mongoConfigs"
+	backupAdministratorDaemonBasePath                        = "admin/backup/daemon/configs"
+	backupAdministratorProjectJobBasePath                    = "admin/backup/groups"
+)
+
+// BackupAdministratorService is an interface for using the Backup Administrator
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/nav/administration-backup/
+type BackupAdministratorService interface {
+	BlockstoreService
+	FileSystemStoreService
+	S3BlockstoreService
+	OplogService
+	SyncService
+	DaemonService
+	ProjectJobService
+}
```

Comment:
I think I know what you were trying to do here but having this many levels of services is not something we've done before

Reply:
See [Projects and users](https://docs.opsmanager.mongodb.com/current/reference/api/nav/groups-and-users/) or [API Keys](https://docs.opsmanager.mongodb.com/current/reference/api/nav/prog-api-keys/)